internal/nerdstorage/command_collection.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/newrelic/newrelic-client-go/newrelic"
+	"github.com/newrelic/newrelic-client-go/pkg/nerdstorage"
+)
+
+var cmdCollection = &cobra.Command{
+	Use:     "collection",
+	Short:   "Read, write, and delete NerdStorage collections.",
+	Example: "newrelic nerdstorage collection --help",
+	Long:    `Read write, and delete NerdStorage collections`,
+}
+
+var cmdCollectionGet = &cobra.Command{
+	Use:   "get",
+	Short: "Retrieve a NerdStorage collection.",
+	Long: `Retrieve a NerdStorage collection
+
+Retrieve a NerdStorage collection.  Valid scopes are ACCOUNT, ENTITY, and USER.
+ACCOUNT scope requires a valid account ID and ENTITY scope requires a valid entity
+GUID.  A valid Nerdpack package ID is required.
+`,
+	Example: `newrelic nerdstorage collection get --scope ACCOUNT --packageId b0dee5a1-e809-4d6f-bd3c-0682cd079612 --accountId 12345678 --collection myCol`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client.WithClient(func(nrClient *newrelic.NewRelic) {
+			var resp []interface{}
+			var err error
+
+			input := nerdstorage.GetCollectionInput{
+				PackageID:  packageID,
+				Collection: collection,
+			}
+
+			switch strings.ToLower(scope) {
+			case "account":
+				if accountID == 0 {
+					log.Fatal("account ID is required when using account scope")
+				}
+
+				resp, err = nrClient.NerdStorage.GetCollectionWithAccountScope(accountID, input)
+			case "entity":
+				if entityGUID == "" {
+					log.Fatal("entity GUID is required when using entity scope")
+				}
+
+				resp, err = nrClient.NerdStorage.GetCollectionWithEntityScope(entityGUID, input)
+			case "user":
+				resp, err = nrClient.NerdStorage.GetCollectionWithUserScope(input)
+			default:
+				log.Fatal("scope must be one of ACCOUNT, ENTITY, or USER")
+			}
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			json, err := prettyjson.Marshal(resp)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			log.Info("success")
+			fmt.Println(string(json))
+		})
+	},
+}
+
+var cmdCollectionDelete = &cobra.Command{
+	Use:   "delete",
+	Short: "Delete a NerdStorage collection.",
+	Long: `Delete a NerdStorage collection
+
+Delete a NerdStorage collection.  Valid scopes are ACCOUNT, ENTITY, and USER.
+ACCOUNT scope requires a valid account ID and ENTITY scope requires a valid entity
+GUID.  A valid Nerdpack package ID is required.
+`,
+	Example: `newrelic nerdstorage collection delete --scope ACCOUNT --packageId b0dee5a1-e809-4d6f-bd3c-0682cd079612 --accountId 12345678 --collection myCol`,
+	Run: func(cmd *cobra.Command, args []string) {
+		client.WithClient(func(nrClient *newrelic.NewRelic) {
+			var err error
+
+			input := nerdstorage.DeleteCollectionInput{
+				PackageID:  packageID,
+				Collection: collection,
+			}
+
+			switch strings.ToLower(scope) {
+			case "account":
+				if accountID == 0 {
+					log.Fatal("account ID is required when using account scope")
+				}
+
+				_, err = nrClient.NerdStorage.DeleteCollectionWithAccountScope(accountID, input)
```

Comment:
What is the field that is being discarded?  We are currently (globally) inconsistent with what we return to users, do we want to show what was deleted (if that is the result), otherwise why does Delete* return a value?

Reply:
The delete API returns a value to let you know if something has been deleted or not.  If you try to delete something that doesn't exist, you'll get a successful call that tells you nothing has been deleted.  I couldn't see the use case for it here.